pkg/k8s/service_cache.go

```diff
@@ -132,22 +137,72 @@ func (s *ServiceCache) UpdateService(k8sSvc *types.Service) ServiceID {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	if oldService, ok := s.services[svcID]; ok {
-		if oldService.DeepEquals(newService) {
+	oldService, ok := s.services[svcID]
+	if ok && oldService.DeepEquals(newService) {
+		return svcID
+	}
+
+	// If the new service has received externalIPs then we need to check
+	// if it previously existed without externalIPs or vice-versa.
+	oldSvcID := svcID
+	if newService.IsK8sExternal() {
+		oldSvcID.k8sExternal = false
```

Comment:
I don't fully understand why we have to set `k8sExternal = false` and vice versa if the new service has external Endpoints or not. What utility does switching the values serve?

Reply:
If it was previously stored *without* or *with* external IPs then the key, of which `k8sExternal` is an attribute of, will not exist in the `s.services` map.

Reply:
Ah, okay. This makes the code harder to read - what about having two separate keys (one for external, one without) instead of swapping the values of the fields? I guess this would use more memory, though, but would make the code more readable.